src/spawn_stubs.c

```diff
@@ -488,6 +502,188 @@ CAMLprim value spawn_unix(value v_env,
     }
   }
 
+  if (new_mask || old_mask)
+    pthread_sigmask(SIG_SETMASK, new_mask, old_mask);
+}
+
+#if defined(USE_POSIX_SPAWN)
+
+CAMLprim value spawn_unix(value v_env,
+                          value v_cwd,
+                          value v_prog,
+                          value v_argv,
+                          value v_stdin,
+                          value v_stdout,
+                          value v_stderr,
+                          value v_use_vfork,
+                          value v_setpgid,
+                          value v_sigprocmask)
+{
+  CAMLparam4(v_env, v_cwd, v_prog, v_argv);
+  CAMLlocal1(e_arg);
+  e_arg = Nothing;
+
+  pid_t pid;
+  int tmp_fds[3] = {0};  // invariant: initialized > 2
+
+  int e_error;
+  char *e_function = NULL;
+
+  posix_spawn_file_actions_t actions;
+  if (posix_spawn_file_actions_init(&actions))
+    caml_raise_out_of_memory();
```

Comment:
Maybe it makes sense to use `unix_error` for this error as well (as for others), to be more informative, since in my experience kernel tends to return unexpected errors in unexpected places.

Reply:
Done (though possibly irrelevant if I rewrite this code in OCaml).